packages/devtools_app/lib/src/screens/memory/panes/diff/controller/simple_controllers.dart

```diff
@@ -14,20 +15,32 @@ class RetainingPathController {
   final invert = ValueNotifier<bool>(true);
 }
 
+class ClassFilterController {
```

Comment:
This isn't really a controller, but rather is a simple data structure. I will push back here and suggest that this class isn't necessary at all. IMO this is overcomplicating something that could easily be declarative when the ClassFilterButton widget is created:

```
ClassFilterButton(
  filter: controller.core.classFilter,
  onChanged: controller.applyFilter,
  rootPackage: serviceManager.rootInfoNow().package,
);
```

This is much more terse and is easier to follow.

Reply:
These three data items are carried together through places, where we do not want widgets to be created (DiffPaneController). So, we need to bundle them together. But, I agree, they are not controllers. 

How about renaming the class to ClassFilterData?

Reply:
Discussed offline.